Sources/ScipioKit/Producer/PIF/FrameworkModuleMapGenerator.swift

```diff
@@ -64,8 +65,20 @@ struct ModuleMapGenerator {
                 + ["}"])
                 .joined()
             case .umbrellaDirectory(let directoryPath):
-                let headers = try walkDirectoryContents(of: directoryPath.scipioAbsolutePath)
-                let declarations = headers.map { "    header \"\($0)\"" }
+                let allHeaders = try headers(under: directoryPath.scipioAbsolutePath)
+
+                let targetDescription = descriptionPackage.targetDescription(of: clangTarget.name)
+
+                let includingHeaders: Set<AbsolutePath> = if let targetDescription {
+                    try Self.excludePaths(
+                        from: allHeaders,
+                        excludedFiles: Set(targetDescription.exclude.map { try RelativePath(validating: $0) }),
+                        targetRoot: clangTarget.path.scipioAbsolutePath
+                    )
+                } else {
+                    allHeaders
+                }
+                let declarations = includingHeaders.map { "    header \"\($0.basename)\"" }
```

Comment:
Just a question, is Tab necessary?

Reply:
Yes, it will be parts of generated modulemap

```modulemap
framework module Foo {
    header "aaa.h"
}
```